fiftyone/core/odm/database.py

```diff
@@ -209,39 +210,47 @@ def _async_connect():
         )
 
 
-def delete_non_persistent_datasets_if_allowed():
-    """Deletes all non-persistent datasets if and only if we are the only
-    client currently connected to the database.
-    """
-    try:
-        num_connections = len(
-            list(
-                _client.admin.aggregate(
-                    [
-                        {"$currentOp": {"allUsers": True}},
-                        {"$project": {"appName": 1, "command": 1}},
-                        {
-                            "$match": {
-                                "appName": foc.DATABASE_APPNAME,
-                                "command.ismaster": 1,
-                            }
-                        },
-                    ]
-                )
-            )
-        )
-    except:
-        logger.warning(
-            "Skipping automatic non-persistent dataset cleanup. This action "
-            "requires read access of the 'admin' database"
-        )
+def _track_db_connection(config, client):
+    # Using an env variable so that this only happens once per "main" process
+    if config.disable_automatic_dataset_cleanup or os.environ.get(
```

Comment:
suggestion: use `os.environ.get("__FIFTYONE_CONN_TRACKER__") != "1")` here to avoid unexpected behavior if someone thinks that they can, for example, set `__FIFTYONE_CONN_TRACKER__ = 0` to have the same effect as unsetting the env var

Reply:
Good catch. I agree specific is better in this case.